src/app/treelist-checkbox/treelist-checkbox.component.html

```diff
@@ -0,0 +1,30 @@
+<ng-container
+  *ngTemplateOutlet="
+    datasetNodeTemplate;
+    context: { children: data !== undefined ? data.children : null, depth: 0 }
+  "></ng-container>
+
+<ng-template #datasetNodeTemplate let-nodes="children" let-depth="depth">
+  <div *ngFor="let node of nodes">
+    <div class="dropdown" style="width: 15px">
+      <button
+        class="{{ hide.value !== 'true' ? 'dropdown-toggle dropdown dropdown-toggle-after' : 'dropdown-toggle' }}"
+        #hide
+        [hidden]="node.children.length === 0"
+        [attr.data-button-id]="node.dataset.id"
+        (click)="updateFilters($event)"></button>
+    </div>
+    <input
+      [ngModel]="this.selectedStudies.has(node.dataset.id)"
+      class="form-check-input"
+      id="{{ node.dataset.id }}"
+      [attr.data-button-id]="node.dataset.id"
```

Comment:
Why not simply use the normal `id` instead of setting an additional attribute with that same value?